packages/core/upload/server/services/metrics.js

```diff
@@ -0,0 +1,86 @@
+'use strict';
+
+const { scheduleJob } = require('node-schedule');
+const { FOLDER_MODEL_UID, FILE_MODEL_UID } = require('../constants');
+
+const rand = max => Math.floor(Math.random() * max);
+const getCronRandomHour = () => `${rand(60)} ${rand(60)} ${rand(24)} ${rand(7)} * *`;
+
+module.exports = ({ strapi }) => {
+  const crons = [];
+  let started = false;
+
+  return {
+    async computeMetrics() {
+      // Folder metrics
+      const pathColName = strapi.db.metadata.get(FOLDER_MODEL_UID).attributes.path.columnName;
+      const folderTable = strapi.getModel(FOLDER_MODEL_UID).collectionName;
+
+      let keepOnlySlashesSQLString = '??';
+      let queryParams = [pathColName];
+      for (let i = 0; i < 10; i += 1) {
+        keepOnlySlashesSQLString = `REPLACE(${keepOnlySlashesSQLString}, ?, ?)`;
+        queryParams.push(String(i), '');
+      }
+
+      const knex = strapi.db.connection;
+      const folderLevelsArray = (
+        await knex(folderTable)
+          .select(
+            knex.raw(
+              `LENGTH(${keepOnlySlashesSQLString}) as depth, count(*) as occurence`,
+              queryParams
+            )
+          )
+          .groupBy('depth')
```

Comment:
Could you try and explain me what this part of the code does?

Reply:
Yes :)
In order to get the depth level of a folder, I take their path and remove all numbers and count (using `LENGTH`) the remaining `/` (I couldn't find a better way to do it than nested `REPLACE`).
I have a row per folder, with their depth. Then, by grouping by their depth and using `count(*)`, it gives me, for each depth, the number of folders that have this depth.

| depth  | occurence |
| ------ | ----------- |
| 1  | 4  |
| 2  | 2 |
| 3  | 5 |

In the end the query returns:
```
[
  { depth: 1, occurence: 4 },
  { depth: 2, occurence: 2 },
  { depth: 3, occurence: 5 },
]
```

Do you think I should write all of this in a comment?

Reply:
The SQL query looks like that:
```
'LENGTH(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(??, ?, ?), ?, ?), ?, ?), ?, ?), ?, ?), ?, ?), ?, ?), ?, ?), ?, ?), ?, ?)) as depth, count(*) as occurence'
```
It's because the `REPLACE` function cannot take a regex, only a string. I want to remove all numbers so I call the `REPLACE` function 10 times, one for each number